development/UserPass.php

```diff
@@ -1,91 +1,114 @@
 <?php
 
+/**
+ * Modified from origin: exampleauth/src/Auth/Source/UserPass.php
+ * 2024-06-19 -- Merged with simplesamlphp 2.2.2, lines/sections marked with GTIS are modified
```

Comment:
@hobbitronics see this comment "lines/sections marked with GTIS are modified"

As a side note, one of my goals is to get rid of all of the places where we do this. It's a maintenance headache that comes up whenever we update SimpleSAMLphp versions. It can be done other ways: Unix tools like sed/awk, diff patches, PHP class inheritance (if possible), or best of all, submitting pull requests to the simplesamlphp/simplesamlphp repository..